docs/core/entity.md

```diff
@@ -119,6 +119,17 @@ The following properties are also available on entities. However, they are for a
 | state_attributes                | <code>dict &#124; None</code> | `None` | State attributes of a base domain. This property is implemented by the domain base entity and should not be implemented by integrations.
 | unit_of_measurement             | <code>str &#124; None</code> | The unit of measurement that the entity's state is expressed in. In most cases, for example for the `number` and `sensor` domains, this is implemented by the domain base entity and should not be implemented by integrations.
 
+## State attributes
+
+Entities can provide additional state attributes besides the built-in ones, to provide further details to its state.
+This should be done by providing a dictionary with keys and values to the `extra_state_attributes` property.
+
+Providing state attributes comes with the following rules:
+
+- If an attribute is expected, it should be in the dictionary. Attributes should not "come and go".
```

Comment:
Should we also indicate our rules for what belongs in state attributes and what should be separate entities and what should be action calls?

Reply:
I think that would probably be good although do we really have clear rules?
I think we could do that in a follow-up in that case.

Reply:
Frenck once mentioned this in the members channel

> Some helpful guidance with that, that I apply (for myself)
> - Would someone extract it using a template into an other (binary) sensor -> Not an attribute
> - It is essential to automation on? Can it be standalone as its own entity? -> Not an attribute
> - It is static, e.g., from configuration; doesn't belong in the state machine -> Not an attribute
> - Do we want to track long term stats of the value? -> Not an attribute
> 
> Additionally:
> - Does is have historical value? If not, exlude it from being recorded.

Reply:
I think that makes sense but I'm not sure we can classify it as rules?
Maybe as "guidelines"?